diff --git a/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldConstraint.php b/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldConstraint.php
index a1eed8d..ca42d7b 100644
--- a/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldConstraint.php
+++ b/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldConstraint.php
@@ -17,6 +17,13 @@ class UniqueFieldConstraint extends Constraint {
   public $message = 'A @entity_type with @field_name %value already exists.';
 
   /**
+   * The name of the property to check.
+   *
+   * @var string
+   */
+  public $propertyName = NULL;
+
+  /**
    * Returns the name of the class that validates this constraint.
    *
    * @return string
diff --git a/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldValueValidator.php b/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldValueValidator.php
index e965310..ec47f43 100644
--- a/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldValueValidator.php
+++ b/core/lib/Drupal/Core/Validation/Plugin/Validation/Constraint/UniqueFieldValueValidator.php
@@ -23,6 +23,14 @@ public function validate($items, Constraint $constraint) {
     $entity_type_id = $entity->getEntityTypeId();
     $id_key = $entity->getEntityType()->getKey('id');
 
+    $propertyName = $constraint->propertyName ?? $item->mainPropertyName() ?? 'value';
+
+    if (!isset($item->getProperties()[$propertyName])) {
+      throw new \LogicException(
+        sprintf('The specified property "%s" does not exist in the field.', $constraint->propertyName)
+      );
+    }
+
     $query = \Drupal::entityQuery($entity_type_id)
       ->accessCheck(FALSE);
 
@@ -34,14 +42,17 @@ public function validate($items, Constraint $constraint) {
     }
 
     $value_taken = (bool) $query
-      ->condition($field_name, $item->value)
+      ->condition(
+        $field_name . '.' . $propertyName,
+        $item->$propertyName
+      )
       ->range(0, 1)
       ->count()
       ->execute();
 
     if ($value_taken) {
       $this->context->addViolation($constraint->message, [
-        '%value' => $item->value,
+        '%value' => $item->$propertyName,
         '@entity_type' => $entity->getEntityType()->getSingularLabel(),
         '@field_name' => $items->getFieldDefinition()->getLabel(),
       ]);
diff --git a/core/tests/Drupal/KernelTests/Core/Validation/UniqueFieldConstraintTest.php b/core/tests/Drupal/KernelTests/Core/Validation/UniqueFieldConstraintTest.php
index 4a77345..262241b 100644
--- a/core/tests/Drupal/KernelTests/Core/Validation/UniqueFieldConstraintTest.php
+++ b/core/tests/Drupal/KernelTests/Core/Validation/UniqueFieldConstraintTest.php
@@ -3,6 +3,7 @@
 namespace Drupal\KernelTests\Core\Validation;
 
 use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Field\BaseFieldDefinition;
 use Drupal\entity_test\Entity\EntityTest;
 use Drupal\entity_test\Entity\EntityTestStringId;
 use Drupal\KernelTests\KernelTestBase;
@@ -23,6 +24,8 @@ class UniqueFieldConstraintTest extends KernelTestBase {
     'entity_test',
     'unique_field_constraint_test',
     'user',
+    'link',
+    'system',
   ];
 
   /**
@@ -148,4 +151,50 @@ public function testViolationDespiteNoAccess() {
     $this->assertEquals($message, $violations[0]->getMessage());
   }
 
+  /**
+   * Tests cases where the validation raises violations for custom properties.
+   *
+   * @covers ::validate
+   */
+  public function testEntityWithFieldCustomProperties() {
+    $value = $this->randomString();
+    $uri1 = 'http://example.com/page1';
+    $uri2 = 'http://example.com/page2';
+
+    $definitions['link'] = BaseFieldDefinition::create('link')
+      ->setLabel('Link')
+      ->addConstraint('UniqueField', ['propertyName' => 'uri']);
+
+    $this->container->get('state')->set('entity_test.additional_base_field_definitions', $definitions);
+    $this->installEntitySchema('entity_test');
+
+    EntityTest::create([
+      'link' => ['title' => $value, 'uri' => $uri1],
+    ])->save();
+
+    // Create a new entity with the same title and different uri value.
+    $entity = EntityTest::create([
+      'link' => ['title' => $value, 'uri' => $uri2],
+    ]);
+    $this->assertCount(0, $entity->name->validate());
+
+    // Create a new entity with the same title and uri value.
+    $entity = EntityTest::create([
+      'link' => ['title' => $value, 'uri' => $uri1],
+    ]);
+
+    $message = new FormattableMarkup('A @entity_type with @field_name %value already exists.', [
+      '%value' => $uri1,
+      '@entity_type' => $entity->getEntityType()->getSingularLabel(),
+      '@field_name' => 'Link',
+    ]);
+
+    /** @var \Symfony\Component\Validator\ConstraintViolationList $violations */
+    $violations = $entity->get('link')->validate();
+
+    // Check that the validation has created the appropriate violation.
+    $this->assertCount(1, $violations);
+    $this->assertEquals($message, $violations[0]->getMessage());
+  }
+
 }
