diff --git a/src/Internal/FileSyncer/Service/FileSyncer.php b/src/Internal/FileSyncer/Service/FileSyncer.php
index 25a9cfff..129f7413 100644
--- a/src/Internal/FileSyncer/Service/FileSyncer.php
+++ b/src/Internal/FileSyncer/Service/FileSyncer.php
@@ -55,7 +55,7 @@ public function sync(
         $this->assertSourceAndDestinationAreDifferent($source, $destination);
         $this->assertSourceIsValid($source);

-        $this->runCommand($exclusions, $source, $destination, $callback);
+        $this->runCommand($exclusions, $source, $destination, $callback, $timeout);
     }

     /** @infection-ignore-all This only makes any difference on Windows, whereas Infection is only run on Linux. */
@@ -113,6 +113,7 @@ private function runCommand(
         PathInterface $source,
         PathInterface $destination,
         ?OutputCallbackInterface $callback,
+        int $timeout,
     ): void {
         $this->ensureDestinationDirectoryExists($destination);
         $command = $this->buildCommand($source, $destination, $exclusions);
@@ -123,6 +124,7 @@ private function runCommand(
                 $this->pathFactory->create('/', $source),
                 [],
                 $callback,
+                $timeout,
             );
         } catch (ExceptionInterface $e) {
             throw new IOException($e->getTranslatableMessage(), 0, $e);
