diff --git a/src/Prophecy/Doubler/Generator/ClassMirror.php b/src/Prophecy/Doubler/Generator/ClassMirror.php
index 6b21623b..63663a5b 100644
--- a/src/Prophecy/Doubler/Generator/ClassMirror.php
+++ b/src/Prophecy/Doubler/Generator/ClassMirror.php
@@ -151,6 +151,10 @@ private function reflectMethodToNode(ReflectionMethod $method, Node\ClassNode $c
             $returnTypes = $this->getTypeHints($method->getReturnType(), $method->getDeclaringClass(), $method->getReturnType()->allowsNull());
             $node->setReturnTypeNode(new ReturnTypeNode(...$returnTypes));
         }
+        elseif (method_exists($method, 'hasTentativeReturnType') && $method->hasTentativeReturnType()) {
+            $returnTypes = $this->getTypeHints($method->getTentativeReturnType(), $method->getDeclaringClass(), $method->getTentativeReturnType()->allowsNull());
+            $node->setReturnTypeNode(new ReturnTypeNode(...$returnTypes));
+        }

         if (is_array($params = $method->getParameters()) && count($params)) {
             foreach ($params as $param) {
@@ -196,7 +200,7 @@ private function hasDefaultValue(ReflectionParameter $parameter)
             return true;
         }

-        return $parameter->isOptional() || ($parameter->allowsNull() && $parameter->getType());
+        return $parameter->isOptional() || ($parameter->allowsNull() && $parameter->getType() && \PHP_VERSION_ID < 80100);
     }

     private function getDefaultValue(ReflectionParameter $parameter)
diff --git a/src/Prophecy/Prophecy/MethodProphecy.php b/src/Prophecy/Prophecy/MethodProphecy.php
index c2c1913d..c45c281d 100644
--- a/src/Prophecy/Prophecy/MethodProphecy.php
+++ b/src/Prophecy/Prophecy/MethodProphecy.php
@@ -73,9 +73,16 @@ public function __construct(ObjectProphecy $objectProphecy, $methodName, $argume
             $this->withArguments($arguments);
         }

-        if (true === $reflectedMethod->hasReturnType()) {
+        $hasTentativeReturnType = method_exists($reflectedMethod, 'hasTentativeReturnType')
+            && $reflectedMethod->hasTentativeReturnType();

-            $reflectionType = $reflectedMethod->getReturnType();
+        if (true === $reflectedMethod->hasReturnType() || $hasTentativeReturnType) {
+            if ($hasTentativeReturnType) {
+                $reflectionType = $reflectedMethod->getTentativeReturnType();
+            }
+            else {
+                $reflectionType = $reflectedMethod->getReturnType();
+            }

             if ($reflectionType instanceof ReflectionNamedType) {
                 $types = [$reflectionType];
